gneiss/_formula.py

```diff
@@ -0,0 +1,90 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+import pandas as pd
+import statsmodels.formula.api as smf
+from skbio.stats.composition import ilr
+from gneiss.util import match, match_tips, rename_internal_nodes
+from gneiss._summary import RegressionResults
+from gneiss.balances import balance_basis
+
+
+def _process(table, metadata, tree):
+    """ Matches tips, features and samples between the table, metadata
+    and tree.  This module returns the features and samples that are
+    contained in all 3 objects.
+    """
+    _table, _metadata = match(table, metadata)
+    _table, _tree = match_tips(table, tree)
+    non_tips_no_name = [(n.name is None) for n in tree.levelorder()
+                        if not n.is_tip()]
+
+    if any(non_tips_no_name):
+        _tree = rename_internal_nodes(_tree)
+    return _table, _metadata, _tree
+
+
+def _transform(table, tree):
+    non_tips = [n.name for n in tree.levelorder() if not n.is_tip()]
+    basis, _ = balance_basis(tree)
+
+    mat = ilr(table.values, basis=basis)
+    ilr_table = pd.DataFrame(mat,
+                             columns=non_tips,
+                             index=table.index)
+    return ilr_table, basis
+
+
+def ols(formula, table, metadata, tree, **kwargs):
+    """ Ordinary Least Squares applied to balances.
+
+    A ordinary least square regression is applied to each balance.
+
+    Parameters
+    ----------
+    formula : str
+        Formula representing the statistical equation to be evaluated.
+        These strings are similar to how equations are handled in R.
```

Comment:
This sentence only means something to someone who is familiar with how they're handled in R...